integrations/reporting_channels/gmail/gmail.py

```diff
@@ -50,19 +53,27 @@ def connect(self, raise_errors=True):
         return self.imap
 
 
+    # Add fetch_mailboxes method in the Gmail class
     def fetch_mailboxes(self):
-        response, mailbox_list = self.imap.list()
+        response, data = self.imap.list()
         if response == 'OK':
-            for mailbox in mailbox_list:
-                mailbox_name = mailbox.split('"/"')[-1].replace('"', '').strip()
-                mailbox = Mailbox(self)
-                mailbox.external_name = mailbox_name
-                self.mailboxes[mailbox_name] = mailbox
+            mailboxes = []
```

Comment:
We have a mailboxes dict which we used any reason to declare a separate list and use it

Reply:
I modified the method to collect the mailbox names in append to mailboxes list and returns it without creating Mailbox objects or modifying self.mailboxes.

Lets keep the previous code as mailboxes dict as it is, I had updated the code to  return the list of mailboxe names